docs/source/developer/repomap.rst

```diff
@@ -0,0 +1,447 @@
+
+
+
+Numba Source Code
+-----------------
+
+Numba ships with both the source code and tests in one package.
+
+- ``numba/`` - all of the source code and tests
+
+
+Public API
+''''''''''
+
+These define aspects of the public Numba interface.
+
+- ``numba/decorators.py`` - User-facing decorators for compiling regular functions on the CPU
+- ``numba/extending.py`` - Public decorators for extending Numba (``overload``, ``intrinsic``, etc)
+- ``numba/ccallback.py`` - ``@cfunc`` decorator for compiling functions to a fixed C singature.  Used to make callbacks.
+- ``numba/npyufunc/decorators.py`` - ufunc/gufunc compilation decorators
+- ``numba/config.py`` - Numba global config options and environment variable handling
+- ``numba/annotations`` - Gathering and printing type annotations of Numba IR
+- ``numba/pretty_annotate.py`` - Code highlighting of Numba functions and types (both ANSI terminal and HTML)
+
+
+Dispatching
+'''''''''''
+
+- ``numba/dispatcher.py`` - Dispatcher objects are compiled functions produced by ``@jit``.  A dispatcher has different implementations for different type signatures.
+- ``numba/_dispatcher.{h,c}`` - C interface to C++ dispatcher implementation
+- ``numba/_dispatcherimpl.cpp`` - C++ dispatcher implementation (for speed on common data types)
+
+
+Compiler Pipeline
+'''''''''''''''''
+
+- ``numba/compiler.py`` - Compiler pipelines and flags
+- ``numba/errors.py`` - Numba exception and warning classes 
+- ``numba/ir.py`` - Numba IR data structure objects
+- ``numba/bytecode.py`` - Bytecode parsing and function identity (??)
+- ``numba/interpreter.py`` - Translate Python interpreter bytecode to Numba IR
+- ``numba/analysis.py`` - Utility functions to analyze Numba IR (variable lifetime, prune branches, etc)
+- ``numba/dataflow.py`` - Dataflow analysis for Python bytecode (used in analysis.py)
+- ``numba/controlflow.py`` - Control flow analysis of Numba IR and Python bytecode
+- ``numba/typeinfer.py`` - Type inference algorithm
+- ``numba/transforms.py`` - Numba IR transformations
+- ``numba/rewrites`` - Rewrite passes used by compiler
+- ``numba/rewrites/__init__.py`` - Loads all rewrite passes so they are put into the registry
+- ``numba/rewrites/registry.py`` - Registry object for collecting rewrit passes
+- ``numba/rewrites/ir_print.py`` - Write print() calls into special print nodes in the IR
+- ``numba/rewrites/static_raise.py`` - Convert exceptions with static arguments into special form that can be lowered
+- ``numba/rewrites/macros.py`` - Generic support for macro expansion in the Numba IR
+- ``numba/rewrites/static_getitem.py`` - Rewrite getitem and setitem with constant arguments to allow type inference
+- ``numba/rewrites/static_binop.py`` - Rewrite binary operations (specifically ``**``) with constant arguments so faster code can be generated
+- ``numba/inline_closurecall.py`` - Inline body of closure functions to call site.
```

Comment:
IIRC this also contains code for the "array comprehensions" feature and both reduction inlining and stencil inlining?

Reply:
Added this information to description